src/Repository/BookingCodes.php

```diff
@@ -275,4 +271,39 @@ public static function deleteBookingCodes( $postId = null ) {
 		}
 	}
 
+	/**
+	 * Fetches given amount of booking codes, randomizes them and puts them in an array, tries to omit doubling
+	 *
+	 * @param int $amount
+	 *
+	 * @return array|null $chosenBookingCodes
+	 */
+	private static function fetchBookingCodes( int $amount ): ?array {
+		$bookingCodes      = Settings::getOption( 'commonsbooking_options_bookingcodes', 'bookingcodes' );
+		$bookingCodesArray = array_filter( explode( ',', trim( $bookingCodes ) ) );
+		$bookingCodesArray = $bookingCodesArray_history = array_map( function ( $item ) {
+			return preg_replace( "/\r|\n/", "", $item );
+		}, $bookingCodesArray );
+
+		// Check if codes are available, return false if not
+		if ( ! count( $bookingCodesArray ) ) {
+			return null;
+		}
+
+		$chosenBookingCodes = array();
+
+		for ($i = 0; $i < $amount; $i++){
+
+			if ( empty( $bookingCodesArray) ){
+				$bookingCodesArray = $bookingCodesArray_history;
```

Comment:
Das hast du doch schon in Zeile 284 gemacht. Insofern bringt das hier eigentlich nix. Im Gegenteil, das führt dazu, dass die if Prüfung überflüssig ist.

Reply:
@hansmorb Es sollen auch Codes generiert werden, wenn die Anzahl der zu erzeugenden Codes die Anazhl der vorhandenen Booking-Codes übersteigt. In dem Fall werden Codes dann eben doppelt vergeben. Die Leute würden sonst anfangen ihre Timeframes kürzer anzulegen, und dann für jeden Timeframe trotzdem wieder aus dem gleichen Pool die gleichen Codes generieren. Vielleicht kann ja irgendwie verhindert werden, dass die gleichen Codes nah beieinander liegen.

Grundsätzlich hatte ich aber mal gedacht, dass @markus-mw in der Originalversion doch eh schon ne art Radnomisierung eingebaut hatte. Zumindest klappte das anfangs ganz gut. Was ist denn nun anders, dass wir da eine neue FUnktion etc. brauchen?

Reply:
Das Problem war, wenn ich den randomizer in seiner Funktion richtig verstanden habe, dass er den zufälligen Buchungscode aus dem aktuellen Tag, der gerade abgearbeitet wird, berechnet. Damit wurde bei aufeinanderfolgenden Tagen einfach nur die Liste abgespult. Die von mir vorgeschlagene Funktion generiert zufällig ein neues Array mit beliebig vielen Buchungscodes, die sich auch nach Möglichkeit nicht wiederholen aber einigermaßen zufällig sind.

Reply:
@chriwen Habe das gerade nochmal geprüft und das Verhalten wie du es in #1110 beschreibst existiert schon mindestens seit Version 2.2.3, das ist also anscheinend normales Verhalten. So wie ich das jetzt vorgeschlagen habe wäre das also ein anderes Verhalten als es vorher der Fall war.